### ISSUE TITLE
Invalid format of `AUTHENTICATE` message

### ISSUE THREAD

**Issue by ProgVal** [sopel-irc/sopel#1974]
## Description

Sopel uses a literal `\0` instead of the null byte in AUTHENTICATE payloads.

### Reproduction steps

1. use this config:

```
auth_username = jilles
auth_password = sesame
auth_method = sasl
```

Sopel will then send this to the server:

```
AUTHENTICATE amlsbGVzXDBqaWxsZXNcMHNlc2FtZQ==
```

### Expected behavior

It should send this:

```
AUTHENTICATE amlsbGVzAGppbGxlcwBzZXNhbWU=
```

base64-decoded, they can be represented as Python literals like this, respectively: `"jilles\\0jilles\\0sesame"` and `"jilles\0jilles\0sesame"`.

### Environment
- Sopel `.version`: master
- Sopel installed via: `setup.py install`

###

This is because of this change: https://github.com/sopel-irc/sopel/pull/1928/files#diff-a3d2e4c9269312d687f6dbab17168611fd71629ef36b099985b9ba4bd293c853R838

this is equivalent to replacing `"\0"` with `"\\0"`.

I'm guessing this was to fix some sort of lint warning, but it also changes the behavior. Instead, you should probably use a `bytes` object.

**Comment by Exirel** [comment]
Nice catch!

**Comment by Exirel** [comment]
Should be fixed by #1976, thank you again @ProgVal for the bug report!
